evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -52,7 +59,7 @@ class TimeSeriesImputer(Transformer):
     # Incompatibility: https://github.com/alteryx/evalml/issues/4001
     # TODO: Remove when support is added https://github.com/alteryx/evalml/issues/4014
     _integer_nullable_incompatibilities = ["X", "y"]
-    _boolean_nullable_incompatibilities = ["X", "y"]
```

Comment:
Realized we could remove `X` from the boolean nullable incompatibilities because `interpolate` isn't a valid impute strategy for boolean columns in X (keeping y, though, because we don't stop interpolate from being called on boolean data - we currently convert to float, so I'm keeping that behavior for now - we should deal with that separately https://github.com/alteryx/evalml/issues/4053 and it shouldnt need to be part of the nullable types work)